vpr/src/route/connection_router.cpp

```diff
@@ -531,9 +623,22 @@ void ConnectionRouter<Heap>::timing_driven_add_to_heap(const t_conn_cost_params
         next_ptr->set_prev_edge(from_edge);
         next_ptr->set_prev_node(from_node);
 
+        if (run_rcv && current->path_data) {
+            next_ptr->path_data->backward_cong = next.path_data->backward_cong;
+            next_ptr->path_data->backward_delay = next.path_data->backward_delay;
+
+            next_ptr->path_data->path_rr = current->path_data->path_rr;
+            next_ptr->path_data->edge = current->path_data->path_rr;
```

Comment:
Is this a bug? Did you mean
```next_ptr->path_data->edge = current->path_data->edge;```?

Reply:
Also, I can see that you are copying vectors here. Is it possible to remove the need for copying by having a shared pointer? Also, if these path data are for recording paths, I think it would be more memory efficient to have a path backtracking data structure (rather than copying the whole paths every time you have a new node)
The path backtracking is basically having the current node as input, and having its previous node (on the presumed best path) as its output. Keep on backtracking, then you'll get the path in reversed order).
I could be talking nonsense here(except for the bug), so make your own judgment call.

Reply:
That's usually the right approach Bill. But in this case, we can't assume that we take the shortest path (we won't) and hence we can have many partially overlapping paths on the heap, and a shared single back-pointer per rr_node won't be sufficient to backtrace when we hit the destination.

Reply:
@vaughnbetz That is true to a point.  Each expansion from the current heap object to all other next expansion points share a comment backtrace from this point backward.  I believe rather than having each heap entry manage the full backtrace, it should have an id to the portion of the trace before it.  This dovetails with my overall comment about having a path management object, rather than free vectors.  Each heap entry can then just maintain it's portion of the path, and the id of its parent.  When the heap item is unused, the path management hangs onto the portion of the path until that fork is completely dead, etc.

I think this approach is viable, and significantly cheaper memory wise, and avoids needing to copy vectors like this.